### DISCUSSION TITLE
Arabic/Hebrew Support

### DISCUSSION THREAD

**Discussion by highdefinitionsound** [phcode-dev/phoenix#1676] [Q&A] [answered]
Strangely, Phoenix Code does not support Arabic. I am facing an issue where the Arabic text I am using is appearing in gibberish. Can you please add a support for Arabic/Hebrew?
![Screen Shot 2024-06-11 at 12 23 53 PM](https://github.com/phcode-dev/phoenix/assets/46069997/52602274-b4b8-4cca-9cb3-95707ba662a7)

**Comment by abose** [comment]
Is this is live view? Is arabic text property displayed in the coding surface? Phoenix cide has native support for Arabic/Hebrew.

**Reply by highdefinitionsound** [reply]
Yes. I was using live preview. Unfortunately, Phoenix Code did not process the Arabic text I used with my code. It was basic HTML and CSS with no additional script.

If this helps more, I tried using both the MacOS app and web IDE. Both shows the same issue. Basically, when I apply any CSS rules to the Arabic text, Phoenix Code starts showing gibberish text.
![IMG_0003](https://github.com/phcode-dev/phoenix/assets/46069997/69409ff1-30d1-47fb-9bf2-86839a21a1b7)

**Reply by abose** [reply, marked as answer]
## update
In the next release (available June-last), we have added support for partial html rendering in Arabic/non-English languages too. This should help with your workflow.

## Earlier answer

I think i got what the issue is @highdefinitionsound . The html is not syntactically correct. That is, you need to specify correct `HTML`  `head` and `body` tags. 

Also for arabic, the meta tag should be there to render across browsers consistently.
` <meta charset="utf-8" />`

BTW, Which browser are you using to test? I would recommend using the desktop app in macs.

Please try this code:

```js
<!doctype html>
<html lang="ar" dir="rtl">
    <head>
        <meta charset="utf-8" />
        <link rel="preconnect" href="https://fonts.googleapis.com" />
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
        <link
            href="https://fonts.googleapis.com/css2?family=Noto+Sans+Arabic:wght@100..900&display=swap"
            rel="stylesheet" />
        <style>
            .noto-sans-arabic {
                font-family: "Noto Sans Arabic", sans-serif;
                font-optical-sizing: auto;
                font-weight: <weight>;
                font-style: normal;
                font-variation-settings: "wdth" 100;
            }
        </style>
    </head>
    <body>
        <p class="blog-post-meta"> 1 يناير 2021 بواسطة <a href="#"> Mark </a></p>
    </body>
</html>
```

This should work.

**Reply by highdefinitionsound** [reply]
Thank you so much for the help. That worked perfectly with me! I have been using VS Code alot in the past few months and I gave myself the impression that Phoenix Code works the same way. I usually use code snippets without <!DOCTYPE.. > nor <head> due to the type of my company’s website. We are using Liferay as a web solution that abuses widgets and applets, and it is terrible in every way possible to use with lots of unnecessary JavaScript codes wrongfully used APIs that killed performance.

Anyways, thank you so much.

**Reply by abose** [reply]
@highdefinitionsound In the next release (available June-last), we have added support for partial html rendering in Arabic/non-English languages too. This should help with your workflow. Thanks for reporting the issue!
